### PR TITLE
calc_item_damage_enemy

### PR DESCRIPTION
**Pull request by Brotenko**

### CODE REVIEW

File: src/1A5830.c
Comment (ethteck) [inline src/1A5830.c:75]: These sound names don't really mean anything to me. The sound of fire in what context?

File: src/17FEB0.c
Comment (ethteck) [inline src/17FEB0.c:83]: this should be PANIC() and the rest of the function doesn't need to be an else, it can just come after.

**Review by ethteck — CHANGES_REQUESTED** [review]
I wish we could get rid of the goto before calling this NON_MATCHING, but Iit's not a dealbreaker I guess. I'll look at this one later

File: src/1A5830.c
Comment (Brotenko) [reply src/1A5830.c:75]: They play accordingly to the dealt damage type "element"

Comment (ethteck) [reply src/1A5830.c:75]: then I think maybe FIRE_DAMAGE and SOUND_ELECTRIC or w/e would be better. Or even SOUND_ELEM_FIRE and SOUND_ELEM_ELECTRIC. just to make it clear that these are battle sounds and not just the sound of fire or electricity in some general sense. if the other one wasn't electricity, my b. I forgot what it was and this tiny diff isn't showing me

File: src/17FEB0.c
Comment (JoshDuMan) [inline src/17FEB0.c:217]: This line a bit long

File: src/17FEB0.c
Comment (nanaian) [inline src/17FEB0.c:116]: This and the other status checks (e.g. transStatus) should use the Statuses enum, i.e.
```suggestion
    if (targetActor->stoneStatus == STATUS_STONE) {
```

File: src/17FEB0.c
Comment (nanaian) [inline src/17FEB0.c:185]: Should use Events enum

**Review by nanaian — CHANGES_REQUESTED** [review]

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 27 functions (+0.06%) on `us`.

✅ This PR fixes 5 warnings!

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀 This PR matches 4 functions (+0.01%) on `us`.

File: include/enums.h
Comment (nanaian) [inline include/enums.h:1634]:
```suggestion
    NPC_FLAG_UPSIDE_DOWN             = 0x00000080, ///< Render NPCs upside-down
```

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

File: src/17FEB0.c
Comment (ethteck) [inline src/17FEB0.c:161]: ! for flag checks

File: src/17FEB0.c
Comment (ethteck) [inline src/17FEB0.c:136]: > 99

**Review by ethteck — CHANGES_REQUESTED** [review]
Before we merge this I'll take a look (tomorrow) and if not, I'm happy to merge after these 2 things are addressed

**Review by ethteck — APPROVED** [review]

**Pull request merged**
